src/nn/graph.ts

```diff
@@ -179,56 +157,31 @@ export class MLGraph {
     for (const outputName of Object.keys(outputTensors)) {
       const tensor = outputTensors[outputName] as tf.Tensor;
       const desc = utils.createOperandDescriptorFromTensor(tensor);
-      const resource = outputs[outputName] as ArrayBufferView;
+      let data;
+      if (syncFlag) {
```

Comment:
I don't get what the usage of `syncFlag` is. Both `tensor.dataSync()` and `await tensor.data()` are synced. Can you elaborate a bit?

Reply:
`tensor.data()` is an async function which would be invoked by `context.comput()` async function, 
and `tensor.dataSync()` is a sync function which would be invoked by `context.computSync()` function.

Any suggestions? Thanks @huningxin.

Reply:
> tensor.data() is an async function which would be invoked by context.comput() async function,
and tensor.dataSync() is a sync function which would be invoked by context.computSync() function.

This sounds right.

However, I am just a bit confused by the usage that calls an async call for sync usage.

Reply:
Yes, in this especial case we don't handle with the return promise object, we use re-allocated output buffer resources to receive computed results.

Reply:
The **await** keyword is what causes asynchronousness.  When calling with `syncFlag=true` condition, the **await** isn't present, then the statements are executed synchronously.

Current there's no rejected handler for `tensor.data()` async function, do you mean that we add some rejected handler for it?

Reply:
> The await keyword is what causes asynchronousness.

This is not true. [await](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await) "pauses the execution of its surrounding async function until the promise is settled (that is, fulfilled or rejected)."

> Current there's no rejected handler for tensor.data() async function, do you mean that we add some rejected handler for it?

No. Because tensor.data() returns a promise, if the promise is rejected,  the await will throw the rejection reason.

Reply:
@huningxin Thanks for your reminder.
I've updated PR to align the usage of invoking sync and async functions, please take another look, thanks.